cmd/clusterctl/client/clusterclass_test.go

```diff
@@ -114,6 +115,17 @@ func TestAddClusterClassIfMissing(t *testing.T) {
 			wantClusterClassInTemplate:  true,
 			wantError:                   false,
 		},
+		{
+			name:                        "should add the cluster class from a different namespace to the template if cluster is not initialized",
+			clusterInitialized:          false,
+			objs:                        []client.Object{},
+			targetNamespace:             "ns1",
+			clusterClassNamespace:       "ns2",
```

Comment:
Can we have a similar test when both namespaces are the same, and one where both are empty?